docs/preview/features/secret-store/named-secret-providers.md

```diff
@@ -38,13 +36,14 @@ using Arcus.Security.Core.Providers;
 
 namespace Application
 {
-
     [ApiController]
     public class OrderController : ControllerBase
     {
         public class OrderController(ISecretStore secretStore)
         {
              // Gets the `ISecretProvider` with the matched name (with either using the `ISecretProvider` as return type or your own generic type).
+             // ⚠ The name of the registered secret providers should be unique when retrieving the concrete secret provider; 
+             //         otherwise, an exception will be thrown when you try to access the `GetProvider<>` or `GetCachedProvider<>`.
```

Comment:
Yes, I understand your confusion, but it's still correct.
When you use either `.GetProvider<TConcreteSecretProvider>` or `.GetCachedSecretProvider<TConcreteSecretProvider>`, you still need to have a unique name, because the targetted secret provider will be cast to the concrete implementation.

When multiple providers have the same name, in this case, we can't cast it.

Reply:
Are there use cases in which were we want or need to specify the concrete type when retrieving the provider ?

Reply:
Yes, when storing Azure Key Vault secrets. That's available in the secret provider, but not in the 'general' `ISecretProvider` interface.